### ISSUE TITLE
Replace pymatgen by Brillouin.jl

### ISSUE THREAD

**Issue by mfherbst** [JuliaMolSim/DFTK.jl#483]
As it says, use https://github.com/thchr/Brillouin.jl for the things we currently need pymatgen for.

**Comment by mfherbst** [comment]
The way to do this is to use [`spg_get_dataset`](https://spglib.github.io/spglib/api.html#spg-get-dataset-and-spg-get-dataset-with-hall-number) from spglib, which returns the [`SpglibDataset`](https://spglib.github.io/spglib/dataset.html#spglib-dataset) in which there is the `spacegroup_number`. This can then be used as the `sgnum` argument to call `Brillouin.irrfbz_path` (using the conventional vectors from spglib as the lattice vectors).

**Comment by harrisonlabollita** [comment]
Hey Michael, I can take care of this!

**Comment by mfherbst** [comment]
Awesome. Then go right ahead!

**Comment by mfherbst** [comment]
Thanks very much!

**Comment by harrisonlabollita** [comment]
Hi, 

Just to confirm, you wanted to replace pymatgen in the [band_structure.jl ](https://github.com/JuliaMolSim/DFTK.jl/blob/master/src/postprocess/band_structure.jl) file with features from Brillouin.jl? 

Where there any other places to touch other than there?

Thanks

**Comment by mfherbst** [comment]
Yes, basically the blob to replace is https://github.com/JuliaMolSim/DFTK.jl/blob/master/src/postprocess/band_structure.jl#L17--L34. But you will not only need Brillouin.jl. You also need to add another c-call to `spglib` which returns a struct, so that struct also needs to be wrapped in Julia code.

**Comment by harrisonlabollita** [comment]
Okay, yes! This is all done. I just wanted to confirm before I made a pull request. Thanks

**Comment by mfherbst** [comment]
I left a comment. Does this work without a segfault?

**Comment by harrisonlabollita** [comment]
Okay, yes I see your comment. Yes, that struct worked without a segfault, but it might be better to incorporate the whole dataset. Comparing to the [python interface to spglib](https://github.com/spglib/spglib/blob/develop/python/spglib/spglib.py), there is a function there which returns that dataset as a python dictionary, as well as another function which returns only the space group number (calls the get dataset function then grabs the space group key from there). Would this be better?

**Comment by mfherbst** [comment]
So far our approach to the spglib interface was to keep it as simple as possible. But of course having a way to get the full dataset would be nice. We could also use at other places in DFTK, actually.

**Comment by harrisonlabollita** [comment]
Okay, I'll go ahead and add create the struct to incorporate the entire dataset. Thanks!

**Comment by mfherbst** [comment]
Thanks, sounds good. Just make sure you don't spend more time on interfacing than necessary ... always very tedious.

**Comment by antoine-levitt** [comment]
Awesome. This'll be especially nice for installation on student machines

**Comment by mfherbst** [comment]
:+1:
